src/rc_sup.erl

```diff
@@ -11,21 +11,40 @@ start_link() ->
   supervisor:start_link({local, ?SERVER}, ?MODULE, []).
 
 init([]) ->
-  Epgsql =
-    #{ id       => rc_pgsql,
-       start    => {rc_pgsql, start_link, []},
-       restart  => permanent,
-       shutdown => infinity,
-       type     => worker,
-       modules  => [rc_pgsql]
-     },
-  Eredis =
-    #{ id       => rc_redis,
-       start    => {rc_redis, start_link, []},
-       restart  => permanent,
-       shutdown => infinity,
-       type     => worker,
-       modules  => [rc_redis]
-     },
-  Children = [Epgsql, Eredis],
-  {ok, {{one_for_one, 3, 10}, Children}}.
\ No newline at end of file
+  Drivers  = application:get_env(reconnections, drivers, []),
+  Children = drivers_specs(Drivers),
+  {ok, {{one_for_one, 3, 10}, Children}}.
+
+%% Priv
+
+drivers_specs(Drivers) ->
+  lists:filtermap(fun(Driver) ->
+                    Args = maps:from_list(Driver),
+                    Module = driver_module(Args),
+                    driver_spec(Module, Args)
+                  end, Drivers).
+
+driver_module(Args) ->
+  case Args of
+    #{type := epgsql} -> rc_epgsql;
+    #{type := eredis} -> rc_eredis;
+    #{type := Driver} -> {error, bad_driver_type, Driver};
+    _                 -> {error, missing_driver}
+  end.
+
+driver_spec({error, bad_driver_type, Driver}, _) ->
```

Comment:
if you established the driver is bad before calling this, I'd say you shouldn't even call it